src/mudlet.cpp

```diff
@@ -1741,53 +1741,57 @@ void mudlet::setDockLayoutUpdated(Host* pHost, const QString& name)
     }
 
     auto pD = pHost->mpConsole->mDockWidgetMap.value(name);
-    if (!mHostDockLayoutChangeMap.value(pHost).contains(name) && pD) {
-        pD->setObjectName(QStringLiteral("%1_changed").arg(pD->objectName()));
+    if (Q_LIKELY(pD) && !mHostDockLayoutChangeMap.value(pHost).contains(name)) {
+        pD->setProperty("layoutChanged", QVariant(true));
         mHostDockLayoutChangeMap[pHost].append(name);
-
         mHasSavedLayout = false;
     }
 }
 
 void mudlet::setToolbarLayoutUpdated(Host* pHost, TToolBar* pTB)
 {
-    if (!pHost) {
+    if (!pHost || !pTB) {
         return;
     }
 
+    // Using the operator[] will instantiate an empty QList<TToolBar*> if there
+    // is not already one in existance for that pHost:
     QList<TToolBar*>& mToolbarLayoutUpdateMap = mHostToolbarLayoutChangeMap[pHost];
     if (!mToolbarLayoutUpdateMap.contains(pTB)) {
-        pTB->setObjectName(QString("%1_changed").arg(pTB->objectName()));
+        pTB->setProperty("layoutChanged", QVariant(true));
         mToolbarLayoutUpdateMap.append(pTB);
-
         mHasSavedLayout = false;
     }
 }
 
 void mudlet::commitLayoutUpdates()
 {
-    // commit changes for dockwidget consoles. (user windows)
-    for (Host* pHost : mHostDockLayoutChangeMap.keys()) {
-        QMap<QString, TDockWidget*>& dockWindowMap = pHost->mpConsole->mDockWidgetMap;
-        QList<QString>& mDockLayoutUpdateMap = mHostDockLayoutChangeMap[pHost];
```

Comment:
Using an `m` prefix was confusing the hell out of me here as I was expecting it to be a member of this or another class... :rage:

Reply:
> I think this could have a beneficial effect on a current issue but I can't track the issue name or number down! slightly_frowning_face
> 
> It is the issue about layouts not being saved correctly each time but only working every other time or if some particular GUI loading action (not looking at the map too quickly on a particular OS???)

This is the one your thinking of @SlySven https://github.com/Mudlet/Mudlet/issues/2076

I am going to test this tonight!